lib/pbench/agent/tool_data_sink.py

```diff
@@ -362,23 +377,69 @@ def run(self):
         finally:
             self.logger.info("Bottle web server exited")
 
+    def tm_log_capture(self):
+        """tm_log_capture - capture all logs written by local and remote Tool
+        Meisters through the Redis server into one file.
+        """
+        tm_log_file = self.benchmark_run_dir / "tm" / "tm.logs"
+        with tm_log_file.open("w") as fp:
+            try:
+                for payload in self._tm_logging_chan:
+                    self.logger.debug("payload: %r", payload)
```

Comment:
Irrelevant side gripe, but it really bugs me that we're using libc printf `%` substitution on the agent side and `{}` substitution on the server side, with the implication that it's basically impossible to log correctly in common code.

Besides, using `%` just feels like going back to C, and we haven't been playing in that playground for a long time.

Reply:
I agree.

But fixing that is more invasive than would be worth doing as part of another PR.  So I'd like to get these Tool Meister changes working before I tackle that.

Reply:
See issue https://github.com/distributed-system-analysis/pbench/issues/2025.